test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs

```diff
@@ -353,6 +353,148 @@ public void TensorFlowTransformMNISTConvTest()
+        /// cost = tf.reduce_mean(-tf.reduce_sum(y* tf.log(pred), reduction_indices=1), name="Loss")
+        /// # Gradient Descent
+        /// optimizer = tf.train.GradientDescentOptimizer(learning_rate).minimize(cost, name = "SGDOptimizer")
+        ///
+        /// # Initialize the variables (i.e. assign their default value)
+        /// init = tf.global_variables_initializer()
+        ///
+        /// # Start training
+        /// with tf.Session() as sess:
+        ///
+        ///     # Run the initializer
+        ///     sess.run(init)
+        ///     tf.saved_model.simple_save(
+        ///         sess,
+        ///         export_dir = "./lr",
+        ///         inputs = { "Features" : x},
+        ///         outputs = {"Prediction": pred
+        ///         })
+        /// </summary>
+        [Fact]
+        public void TensorFlowTransformMNISTTemplateTrainingTest()
+        {
+            var model_location = GetDataPath("lr_model");
+            using (var env = new ConsoleEnvironment(seed: 1, conc: 1))
+            {
+                var dataPath = GetDataPath("Train-Tiny-28x28.txt");
+                var testDataPath = GetDataPath("MNIST.Test.tiny.txt");
+
+                // Pipeline
+                var loader = TextLoader.ReadFile(env,
+                new TextLoader.Arguments()
+                {
+                    Separator = "tab",
+                    HasHeader = false,
+                    Column = new[]
+                    {
+                        new TextLoader.Column("Label", DataKind.Num,0),
+                        new TextLoader.Column("Placeholder", DataKind.Num,new []{new TextLoader.Range(1, 784) })
+
+                    }
+                }, new MultiFileSource(dataPath));
+
+                IDataView trans = new CopyColumnsTransform(env, ("Label", "LabelOriginal")).Transform(loader);
+                trans = CategoricalTransform.Create(env, trans, "Label");
+                trans = NormalizeTransform.CreateMinMaxNormalizer(env, trans, "Features", "Placeholder");
+
+                var args = new TensorFlowTransform.TrainingArguments()
+                {
+                    WrokingDir = @"E:\Tensorflow\WorkingDir",
+                    LabeLColumn = "Label",
+                    OptimizationOperation = "SGDOptimizer",
+                    LossOperation = "Loss",
+                    Epoch = 10,
+                    LearningRateOperation = "SGDOptimizer/learning_rate",
+                    LearningRate = 0.001f,
+                    BatchSize = 20
+
+                };
+
+                trans = TensorFlowTransform.Create(env, trans, model_location, new[] { "Prediction" }, new[] { "Features" }, args);
+                trans = new ConcatTransform(env, "Features", "Prediction").Transform(trans);
+                trans = new CopyColumnsTransform(env, ("LabelOriginal", "Label")).Transform(trans);
+
+                var trainer = new LightGbmMulticlassTrainer(env, "Label", "Features");
```

Comment:
>LightGbmMulticlassTrainer [](start = 34, length = 25)

Note:  a trainer on top is actually not a limitation. In Estimator/Transformer world,prediction ipeline will work without adding a trainer at the end. This is just because it being used in direct instantiation case.